src/FakeItEasy.Analyzer/SymbolHelpers.cs

```diff
@@ -18,10 +19,18 @@ internal static class SymbolHelpers
         /// </summary>
         /// <param name="call">The invocation expression that the symbol is required.</param>
         /// <param name="context">Current context.</param>
+        /// <param name="useFirstCandidateIfNotResolved">Use the first candidate if symbol is not resolved.</param>
         /// <returns>The symbol for the invocation call.</returns>
-        internal static IMethodSymbol GetCalledMethodSymbol(InvocationExpressionSyntax call, SyntaxNodeAnalysisContext context)
+        internal static IMethodSymbol GetCalledMethodSymbol(InvocationExpressionSyntax call, SyntaxNodeAnalysisContext context, bool useFirstCandidateIfNotResolved = false)
         {
-            var method = context.SemanticModel.GetSymbolInfo(call).Symbol as IMethodSymbol;
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(call);
+            var symbol = symbolInfo.Symbol;
+            if (symbol == null && useFirstCandidateIfNotResolved)
+            {
+                symbol = symbolInfo.CandidateSymbols.FirstOrDefault();
+            }

```

Comment:
`GetSymbolInfo` fails to resolve the method if the parameters are invalid, which prevented an invalid call to `A.CallTo(...)` from being seen as a method supporting argument constraints.
Fortunately `GetSymbolInfo` suggests possible candidates, so we use that instead.